### DISCUSSION TITLE
Cant run hoppscotch no matter what i try, pnpm install errors

### DISCUSSION THREAD

**Discussion by ayman999uasy** [hoppscotch/hoppscotch#2941] [Q&A] [answered]
using ubuntu 22.04 following readme file for hoppscotch and always gettings these errors after pnpm install:
root@localhost:/var/local/tools/hoppscotch# pnpm install
Scope: all 8 workspace projects
Lockfile is up to date, resolution step is skipped
Packages: +1679
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Packages are hard linked from the content-addressable store to the virtual store.
  Content-addressable store is at: /root/.local/share/pnpm/store/v3
  Virtual store is at:             node_modules/.pnpm
Progress: resolved 1679, reused 1628, downloaded 0, added 1679, done

dependencies:
+ husky 7.0.4
+ lint-staged 12.5.0

devDependencies:
+ @commitlint/cli 16.3.0
+ @commitlint/config-conventional 16.2.4
+ @types/node 17.0.45
+ http-server 14.1.1
+ vite-plugin-dts 2.0.0-beta.3

packages/codemirror-lang-graphql prepare$ rollup -c
│ Browserslist: caniuse-lite is outdated. Please run:
│   npx browserslist@latest --update-db
│   Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
│
│ src/index.js → dist/index.cjs, ./dist...
│ created dist/index.cjs, ./dist in 9s
└─ Done in 12.7s
packages/hoppscotch-data prepare$ pnpm run build:code && pnpm run build:decl
│ > @hoppscotch/data@0.4.4 build:code /var/local/tools/hoppscotch/packages/hoppscotch-data
│ > vite build
│ vite v3.2.4 building for production...
│ transforming...
│ ✓ 287 modules transformed.
│ rendering chunks...
│ dist/hoppscotch-data.js   46.28 KiB / gzip: 13.39 KiB
│ dist/hoppscotch-data.cjs   34.28 KiB / gzip: 12.04 KiB
│ > @hoppscotch/data@0.4.4 build:decl /var/local/tools/hoppscotch/packages/hoppscotch-data
│ > tsc --project tsconfig.decl.json
└─ Done in 19.8s
packages/hoppscotch-ui postinstall$ pnpm run build
[2 lines collapsed]
│ vite v3.2.4 building for production...
│ transforming...
│ [vite:dts] A error occurred when transform code, maybe there are some inertnal bugs.
│ ✓ 729 modules transformed.
│ rendering chunks...
│ dist/style.css     40.35 KiB / gzip: 5.32 KiB
│ dist/index.es.js   195.80 KiB / gzip: 43.45 KiB
│ dist/index.es.js.map 590.52 KiB
│ [vite:dts] Start generate declaration files...
│ [vite:dts] Declaration files built in 7120ms.
└─ Done in 31.5s
. prepare$ husky install
│ husky - Git hooks installed
└─ Done in 318ms
packages/hoppscotch-js-sandbox postinstall$ pnpm run build
[9 lines collapsed]
│ CJS Build start
│ ESM ⚡️ Build success in 78ms
│ CJS ⚡️ Build success in 74ms
│ ESM lib/index.mjs     18.97 KB
│ ESM lib/index.mjs.map 38.94 KB
│ CJS lib/index.js     20.05 KB
│ CJS lib/index.js.map 38.94 KB
│ DTS Build start
│ DTS ⚡️ Build success in 4353ms
│ DTS lib/index.d.ts 2.65 KB
└─ Done in 6.5s
packages/hoppscotch-common postinstall$ pnpm run gql-codegen
│ > @hoppscotch/common@3.0.1 gql-codegen /var/local/tools/hoppscotch/packages/hoppscotch-common
│ > graphql-codegen --require dotenv/config --config gql-codegen.yml dotenv_config_path="../../.env"
│ [STARTED] Parse Configuration
│ [SUCCESS] Parse Configuration
│ [STARTED] Generate outputs
│ [STARTED] Generate src/helpers/backend/graphql.ts
│ [STARTED] Load GraphQL schemas
│ [FAILED] Failed to load schema
│ [STARTED] Generate src/helpers/backend/backend-schema.json
│ [STARTED] Load GraphQL schemas
│ [FAILED] Failed to load schema
│ [SUCCESS] Generate outputs
│ Failed to load schema for "src/helpers/backend/backend-schema.json"
│         Failed to load schema from undefined:
│
│       Unable to find any GraphQL type definitions for the following pointers:
│
│           - undefined
│
│         Error:
│       Unable to find any GraphQL type definitions for the following pointers:
│
│           - undefined
│
│     at prepareResult (/var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-tools+load@7.7.0_graphql@15.8.0/node_modules/@graphql-tools/load/cjs/load-
│     at loadTypedefs (/var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-tools+load@7.7.0_graphql@15.8.0/node_modules/@graphql-tools/load/cjs/load-t
│     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
│     at async loadSchema (/var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-tools+load@7.7.0_graphql@15.8.0/node_modules/@graphql-tools/load/cjs/sc
│     at async loadSchema (/var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-codegen+cli@2.8.0_h5eoywvcjsa4emif44kddonyyu/node_modules/@graphql-code
│     at async /var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-codegen+cli@2.8.0_h5eoywvcjsa4emif44kddonyyu/node_modules/@graphql-codegen/cli/cjs/
│     at async /var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-codegen+cli@2.8.0_h5eoywvcjsa4emif44kddonyyu/node_modules/@graphql-codegen/cli/cjs/
│     at async /var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-codegen+cli@2.8.0_h5eoywvcjsa4emif44kddonyyu/node_modules/@graphql-codegen/cli/cjs/
│     at async Task.run (/var/local/tools/hoppscotch/node_modules/.pnpm/listr2@4.0.5/node_modules/listr2/dist/index.cjs:979:11)
│     at async /var/local/tools/hoppscotch/node_modules/.pnpm/p-map@4.0.0/node_modules/p-map/index.js:57:22
│         GraphQL Code Generator supports:
│           - ES Modules and CommonJS exports (export as default or named export "schema")
│           - Introspection JSON File
│           - URL of GraphQL endpoint
│           - Multiple files with type definitions (glob expression)
│           - String in config file
│         Try to use one of above options and run codegen again.
│
│ Failed to load schema for "src/helpers/backend/backend-schema.json"
│         Failed to load schema from undefined:
│
│       Unable to find any GraphQL type definitions for the following pointers:
│
│           - undefined
│
│         Error:
│       Unable to find any GraphQL type definitions for the following pointers:
│
│           - undefined
│
│     at prepareResult (/var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-tools+load@7.7.0_graphql@15.8.0/node_modules/@graphql-tools/load/cjs/load-
│     at loadTypedefs (/var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-tools+load@7.7.0_graphql@15.8.0/node_modules/@graphql-tools/load/cjs/load-t
│     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
│     at async loadSchema (/var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-tools+load@7.7.0_graphql@15.8.0/node_modules/@graphql-tools/load/cjs/sc
│     at async loadSchema (/var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-codegen+cli@2.8.0_h5eoywvcjsa4emif44kddonyyu/node_modules/@graphql-code
│     at async /var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-codegen+cli@2.8.0_h5eoywvcjsa4emif44kddonyyu/node_modules/@graphql-codegen/cli/cjs/
│     at async /var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-codegen+cli@2.8.0_h5eoywvcjsa4emif44kddonyyu/node_modules/@graphql-codegen/cli/cjs/
│     at async /var/local/tools/hoppscotch/node_modules/.pnpm/@graphql-codegen+cli@2.8.0_h5eoywvcjsa4emif44kddonyyu/node_modules/@graphql-codegen/cli/cjs/
│     at async Task.run (/var/local/tools/hoppscotch/node_modules/.pnpm/listr2@4.0.5/node_modules/listr2/dist/index.cjs:979:11)
│     at async /var/local/tools/hoppscotch/node_modules/.pnpm/p-map@4.0.0/node_modules/p-map/index.js:57:22
│         GraphQL Code Generator supports:
│           - ES Modules and CommonJS exports (export as default or named export "schema")
│           - Introspection JSON File
│           - URL of GraphQL endpoint
│           - Multiple files with type definitions (glob expression)
│           - String in config file
│         Try to use one of above options and run codegen again.
│
│  ELIFECYCLE  Command failed with exit code 1.
└─ Failed in 2.1s at /var/local/tools/hoppscotch/packages/hoppscotch-common
 ELIFECYCLE  Command failed with exit code 1.

**Comment by SonarSaied** [comment, marked as answer]
Getting the same exact error Ubuntu 22.04

**Reply by SonarSaied** [reply]
Ok I figured it out. Go to your rep and there will be .env.example file. copy that file by running 
`cp .env.example .env `
now run the command again to install and it should work. Disable any popup blockers as that caused an issue for me loading. 
First load seemed to be a bit slow but after it was pretty snappy

**Reply by JayeshHire** [reply]
Thanks I was facing same problem while setting up local development environment locally
